src/lib/sidebar/ListMode.svelte

```diff
@@ -40,6 +40,56 @@
       errorFromFile = `Couldn't load scheme from a file: ${err}`;
     }
   }
+
+  function moveSchemeInList(scheme_reference: string, direction: number) {
+    return function () {
+      const currentSchemeOrder = JSON.parse(
+        JSON.stringify(Object.keys($gjSchemeCollection.schemes))
+      );
+      const currentIndex = currentSchemeOrder.indexOf(scheme_reference);
+      if (
+        currentIndex + direction >= 0 &&
+        currentIndex + direction < currentSchemeOrder.length
+      ) {
+        const newSchemeOrder = swapArrayValues(
+          currentSchemeOrder,
+          currentIndex,
+          currentIndex + direction
+        );
+        const newSchemesObject = getReorderedSchemesObject(
+          newSchemeOrder,
+          $gjSchemeCollection.schemes
+        );
+        $gjSchemeCollection.schemes = newSchemesObject;
+        $gjSchemeCollection = $gjSchemeCollection;
+      }
+    };
+  }
+
+  function swapArrayValues(
```

Comment:
Optional: https://www.freecodecamp.org/news/swap-two-array-elements-in-javascript/ "How to Swap Two Array Elements by Destructuring" looks a bit nicer to me, just a one-liner

Reply:
done, just testing